### PR TITLE
Exchange the Ecto.Type example

### PR DESCRIPTION
**Pull request by LostKobrakai**
I'm not fully happy with it by now so I'm totally open on suggestions/improvements.

Regarding #2229

### CODE REVIEW

File: lib/ecto/type.ex [hunk 1]
```diff
@@ -10,55 +10,65 @@ defmodule Ecto.Type do
...
-  If `id` is an integer field, Ecto will fail in the query above
-  because it cannot cast the string to an integer. By using a
-  custom type, we can provide special casting behaviour while
-  still keeping the underlying Ecto type the same:
+      from s in ShortUrl,
+        where: fragment("?->>? ILIKE ?", s.original_url, "port", "443")
 
-      defmodule Permalink do
+  So the custom type does need to handle the conversion from 
+  external data to runtime data ([`cast/1`](#c:cast/1)) as well as
```
Comment (josevalim) [inline lib/ecto/type.ex]: You can write the callbacks as `c:cast/1` and so on and ExDoc will be able to link to it.

File: lib/ecto/type.ex [hunk 2]
```diff
@@ -10,55 +10,65 @@ defmodule Ecto.Type do
...
+      from s in ShortUrl,
+        where: fragment("?->>? ILIKE ?", s.original_url, "port", "443")
 
-      defmodule Permalink do
+  So the custom type does need to handle the conversion from 
+  external data to runtime data ([`cast/1`](#c:cast/1)) as well as 
+  transforming that runtime data into the `:map` Ecto native type and 
+  back ([`dump/1`](#c:dump/1) and [`load/1`](#c:load/1)).
+
+      defmodule URI.Type do
```
Comment (josevalim) [inline lib/ecto/type.ex]: Let's call it `EctoURI`. We should avoid polluting namespaces of modules we don't own. :)

**Comment by josevalim** [comment]
I definitely think this is an improvement over the previous example. I have added some comments, thank you!

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Comment by LostKobrakai** [comment]
Before I forget. As my last pr to the docs wasn't backported into the 2.2 docs I though it would be a genius idea to issue this own against the v2.2 branch, but it's not on the master branch now 🤦‍♂️ . Do you guys have any guidelines on handling documentation improvements, so at best they don't sit in the master branch for a few month until the next mayor release? For this time should I add another pr against master?

**Comment by josevalim** [comment]
Oh, I missed that. Please always send PRs against master and we backport when asked or if necessary.

**Comment by LostKobrakai** [comment]
Ok, will do.

**Comment by josevalim** [comment]
For now no worries, I have already ported it to master!

**Pull request merged**